src/targets/gpu/compile_hip_code_object.cpp

```diff
@@ -176,8 +177,17 @@ operation compile_hip_code_object(const std::string& content, hip_compile_option
     options.params += " -DMIGRAPHX_NLOCAL=" + std::to_string(options.local);
     options.params += " " + join_strings(compiler_warnings(), " ");
     options.params += " -ftemplate-backtrace-limit=0";
+#ifdef MIGRAPHX_USE_HIPRTC
+    options.params += " -DMIGRAPHX_USE_HIPRTC";
+    // remove following three compilation flags for HIPRTC once fixes from hipRTC are available in
+    // rocm release, add -Werror flag for hipRTC when fixes are available
+    options.params += " -Wno-reserved-identifier";
+    options.params += " -Wno-gnu-line-marker";
+    options.params += " -Wno-old-style-cast";
```

Comment:
These flags need to be added to compile_hip.cpp.

Reply:
Also the flags should only be added when `MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS` is set.

Reply:
Moved those flags inside `compile_hip.cpp`. 

Why do we need `MIGRAPHX_ENABLE_HIPRATC_WORKAROUNDS` ?  How that would be different compared to `MIGRAPHX_USE_HIPRTC` ?  We can use `MIGRAPHX_USE_HIPRTC` to gate workarounds instead  of an extra flag.